### DISCUSSION TITLE
Where is the "network" import in 1.21?

### DISCUSSION THREAD

**Discussion by jimkoke** [micropython/micropython#13699] [General]
Resolved - Thonny does not know to install a "W" version for the Pico W.
I'm using Thonny with a Pico 2W which had micropython 1.20.0 installed, my application worked. I copied my entire application back to my Mac, then updated micropython to the "latest" version, **it chose** 1.21.0. I copied my application back to the Pico, ran it, and found that the "network" module cannot be found:

`ImportError: no module named 'network'`

The import statement is:

`import machine, os, time, network, gc`

I cannot find any description of this "problem" and want to assume that it is something that I did, but am loss as to what the problem could be. To be repetitive, this worked perfectly with 1.20.0.

Simple example:

    try:
        import network
    except ImportError as e:
        print("Error: ", e)
    else:
        print("Import worked")

Result:

    Error:  no module named 'network'

**Comment by peterhinch** [comment]
My guess is that you installed firmware for the standard Pico rather than that for the Pico W.

**Reply by jimkoke** [reply]
I'll look into that, but Thonny did not offer a "W" version.

That was the problem - I let Thonny install the upgrade instead of doing it myself. I'll send a note to the "Thonny" people to see if they are considering supporting the "W" directly.

Thanks!

"If you want it done right, do it yourself."

**Reply by mendenm** [reply]
@jimkoke Thonny 4.1.4 for Macintosh seems to know how.
<img width="611" alt="Screenshot 2024-02-18 at 2 16 02 PM" src="https://github.com/micropython/micropython/assets/48418386/a712b491-7ee1-4a59-ab08-1aff09e9564d">

**Reply by peterhinch** [reply]
My first port of call whenever someone mentions Thonny in the context of a problem is to recommend other means.

**Reply by mendenm** [reply]
@peterhinch I sort of *like* Thonny. It has its foibles, but does the job most of the time.  :-)
